### DISCUSSION TITLE
Merge new SSR context on client-side navigation

### DISCUSSION THREAD

**Discussion by DAreRodz** [WordPress/gutenberg#53057] [Interactivity API]
Currently, an element's context (i.e., `wp-context`) is initialized once (during hydration). However, when client-side navigation happens, it could contain a newly defined context for the new HTML that should not be ignored.

We must read and merge the new context into the current one, keeping the existing props.

**Comment by luisherranz** [comment]
Also relevant here: https://github.com/WordPress/gutenberg/discussions/53056#discussioncomment-6567092

**Comment by luisherranz** [comment]
I made a preliminary version of this [in the WooCommerce repo](https://github.com/woocommerce/woocommerce-blocks/pull/10006/commits/4d579a033d9a22bea94931a7c3385867df6ebb56) because they need it.

**Comment by luisherranz** [comment]
I opened a PR to fix this: 

- https://github.com/WordPress/gutenberg/pull/53853

**Reply by luisherranz** [reply]
This has been merged now.
